src/components/AvatarWithImagePicker.js

```diff
@@ -116,166 +116,118 @@ const defaultProps = {
     originalFileName: '',
 };
 
-class AvatarWithImagePicker extends React.Component {
-    constructor(props) {
-        super(props);
-        this.animation = new SpinningIndicatorAnimation();
-        this.setError = this.setError.bind(this);
-        this.isValidSize = this.isValidSize.bind(this);
-        this.showAvatarCropModal = this.showAvatarCropModal.bind(this);
-        this.hideAvatarCropModal = this.hideAvatarCropModal.bind(this);
-        this.state = {
-            isMenuVisible: false,
-            validationError: null,
-            phraseParam: {},
-            isAvatarCropModalOpen: false,
-            imageName: '',
-            imageUri: '',
-            imageType: '',
+function AvatarWithImagePicker(props) {
+    const animation = new SpinningIndicatorAnimation();
+    const [isMenuVisible, setIsMenuVisible] = useState(false);
+    const [validationError, setValidationError] = useState(null);
+    const [phraseParam, setPhraseParam] = useState({});
+    const [isAvatarCropModalOpen, setIsAvatarCropModalOpen] = useState(false);
+    const [imageName, setImageName] = useState('');
+    const [imageUri, setImageUri] = useState('');
+    const [imageType, setImageType] = useState('');
+    const anchorRef = useRef();
+
+    useEffect(() => {
+        if (props.isUploading) {
+            animation.start();
+        }
+        
+        return () => {
+            animation.stop();
         };
-        this.anchorRef = React.createRef();
-    }
+    }, [props.isUploading]);
 
-    componentDidMount() {
-        if (!this.props.isUploading) {
-            return;
+    useEffect(() => {
+        if (!props.isFocused) {
+            setError(null, {});
         }
+    }, [props.isFocused]);
 
-        this.animation.start();
-    }
+    const setError = (error, phraseParam) => {
+        setValidationError(error);
+        setPhraseParam(phraseParam);
+    };
 
-    componentDidUpdate(prevProps) {
-        if (!prevProps.isFocused && this.props.isFocused) {
-            this.setError(null, {});
-        }
-        if (!prevProps.isUploading && this.props.isUploading) {
-            this.animation.start();
-        } else if (prevProps.isUploading && !this.props.isUploading) {
-            this.animation.stop();
-        }
-    }
-
-    componentWillUnmount() {
-        this.animation.stop();
-    }
-
-    /**
-     * @param {String} error
-     * @param {Object} phraseParam
-     */
-    setError(error, phraseParam) {
-        this.setState({validationError: error, phraseParam});
-    }
-
-    /**
-     * Check if the attachment extension is allowed.
-     *
-     * @param {Object} image
-     * @returns {Boolean}
-     */
-    isValidExtension(image) {
-        const {fileExtension} = FileUtils.splitExtensionFromFileName(lodashGet(image, 'name', ''));
+    const isValidExtension = (image) => {
```

Comment:
```suggestion
    /**
     * Check if the attachment extension is allowed.
     *
     * @param {Object} image
     * @returns {Boolean}
     */
    const isValidExtension = (image) => {
```

let's keep the comments